### PR TITLE
Implement undo/redo in view editor

### PR DESCRIPTION
**Pull request by elvisisking**
See commits for more details

### CODE REVIEW

**Comment by elvisisking** [comment]
Reopening to try and figure out what the merge conflicts are.

File: src/main/ngapp/src/app/dataservices/shared/view.model.ts
Comment (mdrillin) [inline src/main/ngapp/src/app/dataservices/shared/view.model.ts:119]: getName() is not adequate for identifying the source, but I think you are aware of that - and we can change to path with subsequent commit

**Review by mdrillin — APPROVED** [review]
I'm fine to merge if you want.  We can update once the path changes are merged.

**Review by phantomjinx — APPROVED** [review]
Looks fine.

**Pull request merged**
